File: CppPractice/Project1/Project1/Leetcode4.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;

class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        vector<int> num_merge;
        num_merge.insert(num_merge.end(), nums1.begin(), nums1.end());
        num_merge.insert(num_merge.end(), nums2.begin(), nums2.end());
        sort(num_merge.begin(), num_merge.end());
        if (num_merge.size()%2==1)
        {
            return (double)num_merge[num_merge.size() / 2];

        }
        else
        {
            return ((double)num_merge[num_merge.size() / 2]+ (double)num_merge[num_merge.size() / 2-1])/2;

        }

    }
};

File: Pythonleetcode/Leetcode4.py
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        nums=sorted(nums1+nums2)
        if len(nums)%2==0:
            ans=(nums[int(len(nums)/2)-1]+nums[int(len(nums)/2)])/2.0
        else:
            ans = nums[int(len(nums)/2)]
        return ans

print(Solution().findMedianSortedArrays(nums1 = [1,3], nums2 = [2]))
print(Solution().findMedianSortedArrays(nums1 = [1,2], nums2 = [3,4]))
